src/screens/SignIn.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
+import { Auth } from 'aws-amplify';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import AuthTextInput from '../components/AuthTextInput';
+import AuthButton from '../components/buttons/AuthButton';
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { useUser } from '../lib/user';
+import { AuthenticationStackParamList } from '../components/navigators/AuthenticationNavigator';
+
+type SignInRouteProp = RouteProp<AuthenticationStackParamList, 'SignIn'>;
+
+const SignIn: React.FC = () => {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const { params } = useRoute<SignInRouteProp>();
+    const user = useUser();
+
+    const navigation = useNavigation();
+
+    async function signIn() {
```

Comment:
same as above... this should have loading functionality so we can disable the button when the asynchronous work is happening, check out the follower button for a good example of how to do this